lore/developers/contributing/designs/network-design.md

```diff
@@ -0,0 +1,76 @@
+---
+name: network design
+layout: default
+---
+
+# BonnyCI Cloud Network Design
+
+## Problem Statement
+
+Before we redeploy BonnyCI to our new cloud environment, we need to rethink our network design choices.  On our existing cloud, we are running all services and hosts on a single flat internal network.  This allows all hosts to communicate with one another, which is less than ideal considering some hosts (the nodepool slaves) will be executing arbitrary code from third party repositories and patches.  We need to design our network topology in a way that provides the bare minimum required connectivity both internally between hosts and from the outside world in.
+
+## Overview
+
+We will utilize two separate internal networks to isolate our production services from slave nodes that will be executing tests.  The two networks will be connected by a L3 router which will provide connectivity between internal networks as well outbound connectivity to the internet.  Security groups will be used to limit connectivity from the outside world, allow connectivity between hosts on the same network, and prohibit connectivity from slave nodes to the rest of the infrastructure.  This network topology will be configured in Neutron via Ansible automation and will be scoped to production project.  It should be trivial for developers to recreate this topology within their own tenant and project.
+
+### Projects
+
+Network resources will be created within a ``control-plane`` project.  We will be utilizing shared networks, which must be created by a user with the ``cloud_admin`` role on that project. Compute resources for control plane services will be created within the ``control-plane`` project.  Nodepool itself will be configured to use a specific ``nodepool`` project for creating its instances with interfaces attached to a shared network that was created on the ``control-plane`` project.
```

Comment:
Inline code in [GitHub Flavored Markdown](https://guides.github.com/features/mastering-markdown/) is done with single backticks.

For example, I would rewrite this line as:

```markdown
Network resources will be created within a `control-plane` project.  We will be utilizing shared networks, which must be created by a user with the `cloud_admin` role on that project. Compute resources for control plane services will be created within the `control-plane` project.  Nodepool itself will be configured to use a specific `nodepool` project for creating its instances with interfaces attached to a shared network that was created on the `control-plane` project.
```